bazel/envoy_build_system.bzl

```diff
@@ -20,12 +20,45 @@ def envoy_copts(repository, test = False):
         repository + "//bazel:dbg_build": ["-ggdb3"],
     }) + select({
         repository + "//bazel:disable_tcmalloc": [],
+        "//bazel/osx:darwin": [],
```

Comment:
I cribbed this from @Reflejo's old branch. I believe he encountered problems with tcmalloc deadlocking (c.f. https://github.com/lyft/envoy/issues/128#issuecomment-281342790).

I'm game to try turning it on.